sync/tests.md

```diff
@@ -0,0 +1,133 @@
+# Testing plan
+
+We'll test two different types of syncs:
+
+* **Fresh sync**: Syncing a course for the very first time. In this case, we just need to ensure that all the correct entities have been created in the DB. In other words, we're only checking that things exist. This will likely be just a single test case that makes a large number of assertions after a single sync.
+
+* **Incremental sync**: Syncing a course that already exists. In this case, we also need to check that the correct things happen in the case of updates and deletions. So we're checking that new things have been created, existing things have been updated, and removed things have been deleted (possible soft-deleted).
+
+The following tables are potentially modified during a sync:
+* `course_instances`
+* `assessments`
+* `assessment_sets`
+* `topics`
+* `tags`
+* `course_instance_access_rules`
+* `assessment_access_rules`
+* `zones`
+* `alternative_groups`
+* `assessment_questions`
+* `questions`
+* `question_tags`
+
+## `infoCourse.json`
+
+### Fresh sync
+* Course is created in the `course_instances` table
+* Assessment sets are created in the `assessment_sets` table
+* Topics are created in the `topics` table
+* Tags are created in the `tags` table
+
+### Incremental sync
+* Course name/title/options are modified
+  * Modification should be reflected in the DB
+
+* An assessment set is added
+  * The assessment set should be added to the `assessment_sets` table
+* An assessment set is modified
+  * The assessment set should be updated in the `assessment_sets` table
+* An assesment set is removed
+  * The assesment set should be deleted from the `assessment_sets` table
+  * Sync should error if any assessments still reference this assessment set
```

Comment:
I guess that an assessmentSet having its `name` changed is a special case of “remove and add” — I’m not sure if that’s worth testing separately.

Reply:
I did add an independent test for this.